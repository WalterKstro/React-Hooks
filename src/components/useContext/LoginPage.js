import { useContext } from "react"
import UserContext from "./UserContext"

const LoginPage = () => {
    const {setState} = useContext(UserContext)
    const user = {
        id: '#5684564654894798',
        username: '@kstro',
        email: 'walterkstro@gmail.com'
    }
    return (
        <div>
            <h1>Login Page</h1>
            <button 
                className="btn btn-primary"
                onClick={() => setState(user)}
                >Login</button>
        </div>
    )
}
    
export default LoginPage
