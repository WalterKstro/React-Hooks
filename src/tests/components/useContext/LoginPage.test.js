import { mount } from "enzyme"
import LoginPage from "../../../components/useContext/LoginPage"
import UserContext from "../../../components/useContext/UserContext"

describe('Testing in component <LoginPage/>', () => {
    let component
    let user
    let setState

    beforeEach(() => {
        user = {
            id: '#5684564654894798',
            username: '@kstro',
            email: 'walterkstro@gmail.com'
        }
        setState = jest.fn()
        
        component = mount(

            <UserContext.Provider value={{
                    setState
                }}>
                    <LoginPage/>
            </UserContext.Provider>
        )
    })

    test('should show correctly the component <LoginPage/>', () => {
          expect(component).toMatchSnapshot()
    })

    test('should call the function setState with the argument user', () => {
        const button = component.find('button').simulate('click')

        expect(setState).toHaveBeenCalledWith(user)
    })
    
    
})
