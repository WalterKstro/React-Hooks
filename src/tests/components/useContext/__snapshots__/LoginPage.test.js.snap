// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component <LoginPage/> should show correctly the component <LoginPage/> 1`] = `
<div>
  <h1>
    Login Page
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
